graphs/minimum_spanning_tree_prims.py

```diff
@@ -47,59 +47,78 @@ def bottom_to_top(val, index, heap, position):
             if val < heap[parent]:
                 heap[index] = heap[parent]
                 position[index] = position[parent]
-                set_position(position[parent], index)
+                self.set_position(position[parent], index)
             else:
                 heap[index] = val
                 position[index] = temp
-                set_position(temp, index)
+                self.set_position(temp, index)
                 break
             index = parent
         else:
             heap[0] = val
             position[0] = temp
-            set_position(temp, 0)
+            self.set_position(temp, 0)
 
-    def heapify(heap, positions):
+    def heapify(self, heap, positions):
         start = len(heap) // 2 - 1
         for i in range(start, -1, -1):
-            top_to_bottom(heap, i, len(heap), positions)
+            self.top_to_bottom(heap, i, len(heap), positions)
 
-    def delete_minimum(heap, positions):
+    def delete_minimum(self, heap, positions):
         temp = positions[0]
         heap[0] = sys.maxsize
-        top_to_bottom(heap, 0, len(heap), positions)
+        self.top_to_bottom(heap, 0, len(heap), positions)
         return temp
 
-    visited = [0 for i in range(len(l))]
-    nbr_tv = [-1 for i in range(len(l))]  # Neighboring Tree Vertex of selected vertex
+
+def prisms_algorithm(adjacency_list):
+    """
+    >>> adjacency_list = {0: [[1, 1], [3, 3]],
```

Comment:
Excellent.  🚀 

Please also try to find better names for `m`, `'n`, `e`, etc.  Thanks much!